datafusion-federation/src/analyzer.rs

```diff
@@ -20,6 +21,13 @@ impl AnalyzerRule for FederationAnalyzerRule {
     // TableScans from the same FederationProvider.
     // There 'largest sub-trees' are passed to their respective FederationProvider.optimizer.
     fn analyze(&self, plan: LogicalPlan, config: &ConfigOptions) -> Result<LogicalPlan> {
+    
+        if !contains_federated_table(&plan)? {
+            return Ok(plan);
+        }
+
+        let plan = optimize_plan(plan)?;
```

Comment:
I don't think this will be something we can upstream as is, we need to think about how to make this generic somehow. Not for this PR.